nx/lib/nx.ex

```diff
@@ -9167,6 +9167,93 @@ defmodule Nx do
     end
   end
 
+  @doc """
+  Finds the variance of a tensor.
+
+  The variance is the average of the squared deviations from the mean.
+  The mean is typically calculated as sum(x) / N, where N = total of elements.
+  If, however, ddof is specified, the divisor N - ddof is used instead.
+
+  ## Examples
+
+  iex> Nx.variance(Nx.tensor([[1, 2], [3, 4]]))
+
+  #Nx.Tensor<f32 1.25>
+
+  iex> Nx.variance(Nx.tensor([[1, 2], [3, 4]]), 1)
+
+  #Nx.Tensor<f32 1.6666666269302368>
+  """
+  @doc type: :aggregation
+  @spec variance(tensor :: Nx.Tensor.t(), ddof :: number()) :: Nx.Tensor.t()
+  def variance(tensor, ddof \\ 0)
+
+  def variance(tensor, ddof) do
+    %T{shape: shape} = tensor = to_tensor(tensor)
+
+    total = Tuple.product(shape)
+
+    mean = mean(tensor)
+
+    tensor
+    |> subtract(mean)
+    |> power(2)
+    |> sum()
+    |> divide(total - ddof)
+  end
+
+  @doc """
+  Finds the standard deviation of a tensor.
+
+  if ddof is specified, the divisor N - ddof is used to calculate the variance.
+
+  ## Examples
+
+  iex> Nx.standard_deviation(Nx.tensor([[1, 2], [3, 4]]))
+
+  #Nx.Tensor<f32 1.1180340051651>
+  """
+  @doc type: :aggregation
+  @spec standard_deviation(tensor :: Nx.Tensor.t(), ddof :: number()) :: Nx.Tensor.t()
+  def standard_deviation(tensor, ddof \\ 0)
```

Comment:
Since we may want to add an `axes` option to those functions, perhaps `ddof` must be an option too? See how `mean` is done, we want to accept the same options. :)